templates/post_buildspec.yml.tpl

```diff
@@ -71,4 +77,16 @@ phases:
                         echo "Not managed development infrastructure"
                         echo "Subscribing log group "${APP_NAME}-${ENV_NAME}" to "$DATADOG_LAMBDA_FUNCTION_ARN""
                 fi
+        fi
+      - |
+        PIPELINE_TYPE=$(consul kv get "terraform/${APP_NAME}/app-env.json" | jq -r ".${ENV_NAME}.pipeline_type"  2>/dev/null)
+        if [ "$PIPELINE_TYPE" == "cd" ]; then
+          SHARED_LAYER=$(consul kv get "terraform/${APP_NAME}/app-env.json" | jq -r ".${ENV_NAME}.env_type"  2>/dev/null)
+          IS_MANAGED_ENV=$(consul kv get "terraform/${APP_NAME}/app-env.json" | jq -r ".${ENV_NAME}.is_managed_env"  2>/dev/null)
+          if [ "$IS_MANAGED_ENV" = "true" ]; then
+            CURRENT_COLOR=$(consul kv get "infra/${APP_NAME}-${ENV_NAME}/current_color" 2>/dev/null)
+          fi
+          aws s3api get-object --bucket s3-codepipeline-backstage-$SHARED_LAYER --key inframap/generator.sh generator.sh
+          chmod +x generator.sh
+          sh generator.sh "${APP_NAME}" "${ENV_NAME}" "$CURRENT_COLOR" "$CONSUL_URI" "$CONSUL_TOKEN"
```

Comment:
Without checking the generator code here - please make sure it has good error handling - we dont want to fail the build in case the inframap step fails for some reason.